src/main/java/org/prebid/server/deals/targeting/RequestContext.java

```diff
@@ -211,14 +212,27 @@ public LookupResult<List<Size>> lookupSizes(TargetingCategory category) {
             throw new TargetingSyntaxException("Unexpected category for fetching sizes for: " + type);
         }
 
-        final List<Format> formats = getIfNotNull(getIfNotNull(imp, Imp::getBanner), Banner::getFormat);
-        final List<Size> sizes = ListUtils.emptyIfNull(formats).stream()
-                .map(format -> Size.of(format.getW(), format.getH()))
-                .toList();
+        final List<Size> sizes = Stream.concat(sizesFromBanner(imp), sizesFromVideo(imp)).toList();
 
         return !sizes.isEmpty() ? LookupResult.ofValue(sizes) : LookupResult.empty();
     }
 
+    private static Stream<Size> sizesFromBanner(Imp imp) {
```

Comment:
Maybe we should use a more standard approach for this method and for `sizesFromVideo`? Return `List` instead of a stream, also update the 215 line to use `ListUtils.union()`?